test/test_exaerror.lua

```diff
@@ -16,15 +16,38 @@ function test_exaerror.test_get_message()
     luaunit.assertEquals(msg:get_message(), "This is a message.")
 end
 
-function test_exaerror.test_get_interpolated_message()
-    local tests = {
-        {message = "Foo '%s'", parameters = {"the a parameter"}, expected = "Foo 'the a parameter'"},
-        {message = 'The %s is "%d".', parameters = {"answer", 42}, expected = 'The answer is "42".'}
-    }
-    for _, test in ipairs(tests) do
-        local msg = exaerror.create("W-BARZOO-2", test.message, unpack(test.parameters))
-        luaunit.assertEquals(msg:get_message(), test.expected)
-    end
+function test_exaerror.test_get_interpolated_message_with_single_string()
+    local msg = exaerror.create("W-BARZOO-2", "Foo {{string}}", {string = "a text"})
+    luaunit.assertEquals(msg:get_message(), "Foo 'a text'")
+end
+
+function test_exaerror.test_get_interpolated_message_with_string_and_number()
+    local msg = exaerror.create("W-BARZOO-2", 'The {{string}} is "{{number}}".', {string = "answer", number = 42})
+    luaunit.assertEquals(msg:get_message(), "The 'answer' is \"42\".")
+end
+
+function test_exaerror.test_get_parameter_description()
+    local msg = exaerror:new({
+        code = "E-A-1",
+        message = "A, {{b}}, {{c}}",
+        parameters = {
+            b = {value = "B", description = "the B"},
+            c = {value = "C", description = "the C"}
+        }
+    })
+    luaunit.assertEquals(msg:get_parameter_description("c"), "the C")
```

Comment:
I don't see why we need this method. It's only required by the crawler. And that will probably use a different class structure.

Reply:
I don't want to let the existence of the parameter description untested. And since Lua is very hard to statically analyze, it could very well turn out that I have to execute code fragments in order to extract the message.